src/plopp/widgets/clip3d.py

```diff
@@ -0,0 +1,397 @@
+    plain_update:
+        A function to update the scene without debouncing.
+    throttled_update:
+        A function to update the scene with debouncing.
+    color:
+        Color of the cut's outline.
+    linewidth:
+        Width of the line delineating the outline.
+    """
+
+    def __init__(
+        self,
+        limits: Tuple[sc.Variable, sc.Variable, sc.Variable],
+        direction: Literal['x', 'y', 'z'],
+        plain_update: Callable,
+        throttled_update: Callable,
+        color: str = 'red',
+        linewidth: float = 1.5,
+    ):
+        self._limits = limits
+        self._direction = direction
+        axis = 'xyz'.index(self._direction)
+        self.dim = self._limits[axis].dim
+        self._unit = self._limits[axis].unit
+        self.visible = True
+        self._plain_update = plain_update
+        self._throttled_update = throttled_update
+
+        w_axis = 2 if self._direction == 'x' else 0
+        h_axis = 2 if self._direction == 'y' else 1
+        width = (self._limits[w_axis][1] - self._limits[w_axis][0]).value
+        height = (self._limits[h_axis][1] - self._limits[h_axis][0]).value
+
+        self.outlines = [
+            p3.LineSegments(
+                geometry=p3.EdgesGeometry(
+                    p3.PlaneBufferGeometry(width=width, height=height)
+                ),
+                material=p3.LineBasicMaterial(color=color, linewidth=linewidth),
+            ),
+            p3.LineSegments(
+                geometry=p3.EdgesGeometry(
+                    p3.PlaneBufferGeometry(width=width, height=height)
+                ),
+                material=p3.LineBasicMaterial(color=color, linewidth=linewidth),
+            ),
+        ]
+        if self._direction == 'x':
+            for outline in self.outlines:
+                outline.rotateY(0.5 * np.pi)
+        if self._direction == 'y':
+            for outline in self.outlines:
+                outline.rotateX(0.5 * np.pi)
+
+        center = [var.mean().value for var in self._limits]
+        vmin = self._limits[axis][0].value
+        vmax = self._limits[axis][1].value
+        dx = vmax - vmin
+        delta = 0.05 * dx
+        self.slider = ipw.FloatRangeSlider(
+            min=vmin,
+            max=vmax,
+            value=[center[axis] - delta, center[axis] + delta],
+            step=dx * 0.01,
+            description=direction.upper(),
+            style={'description_width': 'initial'},
+            layout={'width': '470px', 'padding': '0px'},
+        )
+
+        self.cut_visible = ipw.Button(
+            icon='eye-slash',
+            tooltip='Hide cut',
+            layout={'width': '16px', 'padding': '0px'},
+        )
+
+        for outline, val in zip(self.outlines, self.slider.value):
+            pos = list(center)
+            pos[axis] = val
+            outline.position = pos
+
+        self.unit_label = ipw.Label(f'[{self._unit}]')
+        self.cut_visible.on_click(self.toggle)
```

Comment:
Bug we saw in huddle: When `cut_visible` is toggled the outline is shown even if all outlines are hidden.